### PR TITLE
refactor(topnav): update top navigation component specs

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

`<hx-topnav>` implementing color palette 2.0 styles

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF 2142

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

**Review by 100stacks — CHANGES_REQUESTED** [review]
### @danielmdesigns please see Top Navigation updates based on current specs:

### https://deploy-preview-807--helix-ui.netlify.app/components/topnav

**Review by danielmdesigns — CHANGES_REQUESTED** [review]
Here is some initial feedback:

- Dropshadow for dropdowns are too dark. I would suggest reducing the opacity/alpha transparency there. Selecting this object in Abstract will give you the box-shadow specs that we used in our design
- Nav links should not be in all caps
- Default link colors are wrong. We should use #757575 for all Top & Bottom nav links
- Remove background color from hover. Hover/Selected state link color will be #333333
- Default Icon color should be #9E9E9E (hover/selected icons will be #333333)
- Dropdown links should not have a background on hover. The Default link color will be #333333 with a Hover/Selected color of #970C8D
- Button hover state color is wrong. Button states should be Blue-500 Default, Blue-700 Hover, Blue-900 Selected
- Button should not have border-color on hover

The Navigation link in Abstract has been updated. Please reference this to support the feedback provided above. If you have any other questions, please do not hesitate to ask.

https://share.goabstract.com/cd3268fc-279f-4e26-9fbd-f8abde8330d0

Top Nav Design

**Comment by lalithkarikelli** [comment]
@danielmdesigns :  Please review the changes based on your feedback. 
https://deploy-preview-807--helix-ui.netlify.app/components/topnav
Thanks!

**Review by danielmdesigns — CHANGES_REQUESTED** [review]
Looks great! Thanks for addressing those previous concerns. I believe once the icons are updated (James & I just talked about these) then I believe this will be ready to approve. Great work, thanks for the quick turn around on the updates.

File: docs/components/topnav/index.html [hunk 1]
```diff
@@ -28,86 +28,85 @@ <h3>
...
+                  </footer>
+                </section>
+              </hx-menu>
+            </div>
+          </div>
         </div>
       </nav>
-
+  
+
```
Comment (100stacks) [inline docs/components/topnav/index.html:109]: Please remove extra line.

File: src/scss/components/topnav/index.scss [hunk 2]
```diff
@@ -6,10 +6,18 @@
 /// * ====== Top Navigation  ====== * ///
 /// * ============================= * ///
 
+
```
Comment (100stacks) [inline src/scss/components/topnav/index.scss:9]: Please remove extra blank line.

File: src/scss/components/topnav/index.scss [hunk 3]
```diff
@@ -48,6 +52,13 @@
   .hxTopNavOptionMenu {
     display: flex;
     float: left;
+
+    > hx-disclosure {
+      padding: 0.625rem 1.25rem 0;
+      border-bottom: 2px solid #970c8d;
```
Comment (100stacks) [inline src/scss/components/topnav/index.scss:58]: Please update color to `$purple-500`.

File: src/scss/components/topnav/index.scss [hunk 4]
```diff
@@ -62,63 +73,58 @@
       font-size: 0.813rem;
       height: 3.5rem;
       line-height: 1;
-      padding: 0 0.563rem;
       text-align: center;
       text-decoration: none;
-      text-transform: uppercase;
+      display: flex;
```
Comment (100stacks) [inline src/scss/components/topnav/index.scss:78]: Is this value from the new specs?

File: src/scss/components/topnav/index.scss [hunk 5]
```diff
@@ -62,63 +73,58 @@
...
+        color: $gray-950;
       }
     }
 
     #topnav-menu:hover,                // deprecated
-    #topnavMenuId hx-menuitem:hover {  // deprecated
-      background-color: $gray-900;
-      color: $gray-0 !important;
+    #topnavMenuId hx-menuitem:hover {
+      // deprecated
```
Comment (100stacks) [inline src/scss/components/topnav/index.scss:118]: Please revert this change....move to Line 117.

File: src/scss/components/topnav/index.scss [hunk 6]
```diff
@@ -62,63 +73,58 @@
...
   #topnav-menu,         // deprecated
   #topnav-productMenu,  // deprecated
   .hxTopNavMenu > hx-disclosure {
-    background-color: transparent;
-    color: $gray-400;
-    font-size: 0.875rem;
-    height: 3.5rem;
-    line-height: 1;
     padding: 0.625rem 1.25rem 0;
+    height: 3.5rem;
```
Comment (100stacks) [inline src/scss/components/topnav/index.scss:127]: Please alphabetize these properties.

File: src/scss/components/topnav/index.scss [hunk 7]
```diff
@@ -129,28 +135,39 @@
       width: 0.75rem;
     }
 
+    #user {
+      width: 0.81rem;
+      height: 0.94rem;
+      margin-right: 0.44rem;
+      color: $gray-700;
```
Comment (100stacks) [inline src/scss/components/topnav/index.scss:142]: Please alphabetize these properties.

File: src/scss/components/topnav/index.scss [hunk 8]
```diff
@@ -129,28 +135,39 @@
       width: 0.75rem;
     }
 
+    #user {
```
Comment (100stacks) [inline src/scss/components/topnav/index.scss:138]: Please change to a class:

```css
.hxNavUser {
```

File: src/scss/components/topnav/index.scss [hunk 9]
```diff
@@ -129,28 +135,39 @@
...
   .hxTopNavMenu > hx-menu {
-    background-color: $gray-1000;
-    border: 1px solid $gray-1000;
-    color: $gray-0;
+    background-color: $gray-0;
+    color: $gray-1000;
     min-width: 10.5rem;
     padding: 0;
     top: 14.5rem;
+    box-shadow: 0 0.125rem 0.25rem 0 rgba(0, 0, 0, 0.5);
```
Comment (100stacks) [inline src/scss/components/topnav/index.scss:167]: Please alphabetize these properties.

File: src/scss/components/topnav/index.scss [hunk 10]
```diff
@@ -161,13 +178,12 @@
     }
 
     hx-menuitem:not(.hxMenuKey):hover {
-      background-color: $gray-900;
-      color: $gray-0 !important;
+      color: #970c8d !important;
```
Comment (100stacks) [inline src/scss/components/topnav/index.scss:181]: Do we need the `!important`?

Comment (100stacks) [inline src/scss/components/topnav/index.scss:181]: Please change color from `#970c8d` to `$purple-500`.

**Review by 100stacks — CHANGES_REQUESTED** [review]
### Please see changes below, and squash commits.

File: src/scss/components/topnav/index.scss (hunk 4 above)
Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss:78]: yes, there is a change how navigation icons to be displayed. Based on that change we need display: flex

File: docs/components/topnav/index.html [hunk 11]
```diff
@@ -28,86 +28,85 @@ <h3>
       </h3>
       <br />
     </header>
-
+
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply docs/components/topnav/index.html]: Changes have been done, please review.

File: docs/components/topnav/index.html (hunk 1 above)
Comment (lalithkarikelli) [reply docs/components/topnav/index.html]: Changes have been done, please review.

File: src/scss/components/topnav/index.scss (hunk 2 above)
Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss]: Changes have been done, please review.

File: src/scss/components/topnav/index.scss (hunk 3 above)
Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss]: Changes have been done, please review.

File: src/scss/components/topnav/index.scss (hunk 5 above)
Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss:117]: Changes have been done, please review.

File: src/scss/components/topnav/index.scss (hunk 6 above)
Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss:125]: Changes have been done, please review.

File: src/scss/components/topnav/index.scss (hunk 7 above)
Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss]: Changes have been done, please review.

File: src/scss/components/topnav/index.scss (hunk 8 above)
Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss]: Changes have been done, please review.

File: src/scss/components/topnav/index.scss (hunk 9 above)
Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss]: Changes have been done, please review.

File: src/scss/components/topnav/index.scss (hunk 10 above)
Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss]: Changes have been done, please review.

Comment (lalithkarikelli) [reply src/scss/components/topnav/index.scss]: Changes have been done, please review.

**Review by 100stacks — APPROVED** [review]
### Dev LGTM...great job @lalithkarikelli! 

### @danielmdesigns, we're going to update the icons in a future release.
